test/models/commit_reward_test.rb

```diff
@@ -0,0 +1,49 @@
+require 'test_helper'
+
+class CommitRewardTest < ActiveSupport::TestCase
+  def setup
+    @user1  = create :user, :auth_token => 'abcd1234'
+    @user1.name = 'user1'
+
+    @user2 = create :user, :auth_token => 'abc1234'
+    @user2.name = 'user2'
+
+    @repository1 = create :repository, :language => 'Ruby'
+    @repository2 = create :repository, :language => 'Java'
+
+    3.times.each do |i|
+      commit = create :commit, user: @user1, commit_date: Date.yesterday, repository: @repository1, score: 2, reward: 1
+      @user1.commits << commit
+    end
+
+    2.times.each do |i|
+      commit = create :commit, user: @user2, commit_date: Date.yesterday, repository: @repository2, score: 2.4, reward: 1.6
+      @user2.commits << commit
+    end
+  end
+
+  test 'should check if transactions are created properly' do
+    CommitReward.new.send(:create_transaction)
```

Comment:
call this ``` CommitReward.new.calculate```

Reply:
If I do CommitReward.new.calculate, random values will get assigned to score and reward and I wont be able to check them the way I have checked now.
 Here we just want to check if transaction is being created porperly and badges are updated or not, hence I only called that method.